projects/testing-library/src/lib/testing-library.ts

```diff
@@ -123,14 +123,38 @@ export async function render<SutType, WrapperType = SutType>(
 
   await renderFixture(componentProperties, componentInputs, componentOutputs);
 
+  let renderedPropKeys = Object.keys(componentProperties);
+  let renderedInputKeys = Object.keys(componentInputs);
+  let renderedOutputKeys = Object.keys(componentOutputs);
   const rerender = async (
     properties?: Pick<RenderTemplateOptions<SutType>, 'componentProperties' | 'componentInputs' | 'componentOutputs'>,
   ) => {
-    await renderFixture(
-      properties?.componentProperties ?? {},
-      properties?.componentInputs ?? {},
-      properties?.componentOutputs ?? {},
-    );
+    const newComponentInputs = properties?.componentInputs ?? {};
+    for (const inputKey of renderedInputKeys) {
+      if (!Object.prototype.hasOwnProperty.call(newComponentInputs, inputKey)) {
+        delete (fixture.componentInstance as any)[inputKey];
+      }
+    }
+    setComponentInputs(fixture, newComponentInputs);
+    renderedInputKeys = Object.keys(newComponentInputs);
+
+    const newComponentOutputs = properties?.componentOutputs ?? {};
+    for (const outputKey of renderedOutputKeys) {
+      if (!Object.prototype.hasOwnProperty.call(newComponentOutputs, outputKey)) {
+        delete (fixture.componentInstance as any)[outputKey];
+      }
+    }
+    setComponentOutputs(fixture, newComponentOutputs);
+    renderedOutputKeys = Object.keys(newComponentOutputs);
+
+    const newComponentProps = properties?.componentProperties ?? {};
+    const changes = updateProps(fixture, renderedPropKeys, newComponentProps);
+    if (hasOnChangesHook(fixture.componentInstance)) {
+      fixture.componentInstance.ngOnChanges(changes);
+    }
+    renderedPropKeys = Object.keys(newComponentProps);
+
+    fixture.componentRef.injector.get(ChangeDetectorRef).detectChanges();
```

Comment:
Could you also provide a parameter for this?
With `render` we can disable detectChanges, I suppose some tests would also require this with `rerender`.
Thoughts?

Reply:
Have I understood the concept correctly with https://github.com/testing-library/angular-testing-library/pull/366/commits/4c5a2552e54d73a75d3217c89a35ae0a4bb16655?

I have mixed feelings about it. I want help to provide a nice helpful library. On the other side I am not sure it is a good idea to disable change detection of a component. I just assume the component under test has a design flaw when this is helpful. Or I am just lacking imagination of good use cases.^^